common/rakelib/utils.rake

```diff
@@ -0,0 +1,74 @@
+# Used constants:
+# none
+
+class Utils
+  # formatter types
+  :xcpretty   # pass through xcpretty and store in artifacts
+  :raw        # store in artifacts
+  :to_string  # run using backticks and return output
+
+  # run a command using xcrun and xcpretty if applicable
+  def self.run(cmd, task, subtask = '', xcrun: false, formatter: :raw)
+    commands = xcrun ? [*cmd].map { |cmd|
+      "#{version_select} xcrun #{cmd}"
+    } : [*cmd]
+
+    case formatter
+    when :xcpretty then xcpretty(commands, task, subtask)
+    when :raw then plain(commands, task, subtask)
+    when :to_string then `#{commands.join(' && ')}`
+    else raise "Unknown formatter '#{formatter}'"
+    end
+  end
+
+  # print an info header
+  def self.print_info(str)
+    (red,clr) = (`tput colors`.chomp.to_i >= 8) ? %W(\e[33m \e[m) : ["", ""]
```

Comment:
Fun fact: `\e[33m` isn't actually red, but yellow :wink: maybe we should rename that variable ^^